### ISSUE TITLE
Tabs component isn't enforcing semantic tab functionality - causing accessibility issue

### ISSUE THREAD

**Issue by jaxee** [Shopify/polaris-react#3692]
## Goal

The goal for this issue is to discuss how to prevent the Tabs component from causing an `aria-valid-attr-value` error.

## Problem

The Polaris Tabs component does not force users to use the component in a semantic way. [Tabs are described by WCAG](https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel) as:
```
Tabs are a set of layered sections of content, known as tab panels, that display one panel of content at a time.
Each tab panel has an associated tab element, that when activated, displays the panel.
The list of tab elements is arranged along one edge of the currently displayed panel, most commonly the top edge.
```

The Polaris Tabs component does not force users to provide children for the component. This means that you can use the Polaris Tabs component without having a tab panel. This happens because the `children` prop is optional. Why is this an issue? 

The Tabs component always sets an `aria-controls` value on the tabs themselves. When the component doesn't have children, there isn't an associated id (tab pabel) for the aria-control value. This leads to an `aria-valid-attr-value: Ensure all ARIA attributes have valid values` error.

Error details from an example the Shopify admin:
```
Title: WCAG 4.1.1,WCAG 4.1.2: Ensures all ARIA attributes have valid values (#ALL_SEARCH)
Tags: Accessibility, WCAG 4.1.1, WCAG 4.1.2, aria-valid-attr-value

Issue: Ensures all ARIA attributes have valid values (aria-valid-attr-value - https://dequeuniversity.com/rules/axe/4.0/aria-valid-attr-value?application=msftAI)

Target application: jackie testing ~ Orders ~ Shopify - https://jackie-testing.myshopify.com/admin/draft_orders

Element path: #ALL_SEARCH

Snippet: <a data-polaris-unstyled="true" id="ALL_SEARCH" role="tab" tabindex="0" class="Polaris-Tabs__Tab_375k2 Polaris-Tabs__Tab--selected_11sap" aria-selected="true" aria-controls="ALL_SEARCH-panel" aria-label="All" href="/admin/draft_orders">

How to fix: 
Fix all of the following:
  Invalid ARIA attribute value: aria-controls="ALL_SEARCH-panel"

Environment: Chrome version 87.0.4280.88
```

## Quick look at the code

The [children prop](https://github.com/Shopify/polaris-react/blob/master/src/components/Tabs/Tabs.tsx#L17) is optional.
[If the component has children](https://github.com/Shopify/polaris-react/blob/master/src/components/Tabs/Tabs.tsx#L87), we render the `Panel` component and assign it a correct id (that matches the aria-controls value)
If not, then the[ panel markup is null](https://github.com/Shopify/polaris-react/blob/master/src/components/Tabs/Tabs.tsx#L106).

This causes the `aria-valid-attr-value` error.

Here is where the [aria-controls is set](https://github.com/Shopify/polaris-react/blob/master/src/components/Tabs/components/Tab/Tab.tsx#L106). 

### (1) Conditionally assign the aria-controls attribute 
Assign the aria-controls attribute conditionally. If the component has children, assign the aria-controls attribute and the associated ID on the tab-panel. If the component does not have children, do not assign the aria-controls attribute because we don't have a place to set the associated id.

**Pros**
This fixes the accessibility error without having to drastically refactor the component. This wouldn't be a breaking change.

**Cons**
This exacerbates the `it looks like a tab component but doesn't act like a tab component` semantic issue. Conditionally using the aria attribute abstracts the logic away from the user, which could become confusing unless they look at the source code.

### (2) Have a semantic tabs component and a new component for the non-semantic uses
Have a semantic tab component (with all of the expected behaviour and correct accessibility features) and a visual tabs component (to continue folks to use "tabs" in a more flexible way, without the accessibility for a semantic tabs component).

**Pros**
This fixes the accessibility error and provides alternatives for visual usages of tabs. 

**Cons**
This is considerably more work than the other options but I believe it is the "most correct" way forward. One component can't be accessible and have the flexibility to use tabs semantically and non-semantically. Conflating these two things isn't great. 
Communicating this change to folks that use Polaris would be tough because we'd be taking away flexibility from one component (to make it accessible) and forcing users use a different component. Forcing folks to make a change like this comes with its challenges and push backs.

### (3) Require semantic use of Tabs (children required)
Require the Tabs component to have children, forcing users that implement Tabs to use it in a semantic way.

**Pros**
Fixes the accessibility error because the component would always be able to set the correct id for the tab panel. This would also make it obvious to users HOW to use the component and would prevent mis-use of Tabs.

**Cons**
Takes away flexibility from the component. 
Where this error comes up the most in the admin is on list/index pages (Ex. Draft Orders List, Inventory List, etc.). Tabs are being used as a filtering system, without associated tab panels and the Tabs component lets us do this (causing the accessibility error). This manifests as a list page with filters that have one "tab panel" that gets updated based on a filter when a different tab is chosen. Those cases require a visual "tab" component but not a semantic "tab" component.
